lib/roo_on_rails/logfmt.rb

```diff
@@ -0,0 +1,41 @@
+module RooOnRails
+  # A generator for the logfmt log format.
+  #
+  # @see https://brandur.org/logfmt The original description of logfmt
+  # @see https://godoc.org/github.com/kr/logfmt The 'reference' parser
+  module Logfmt
+    class << self
+      SPACE = 0x20
+      QUOTE = 0x22
+      EQUALS = 0x3d
+
+      def dump(hash)
+        return nil if hash.nil? || hash.empty?
+
+        hash.map { |k, v| "#{k}=#{dump_value(v)}" }.join(' ')
+      end
+
+      private
+
+      def dump_value(v)
+        str = case v
+              when String then v
+              when Symbol then v.to_s
+              else v.respond_to?(:to_json) ? v.to_json : v.inspect
+              end
+        escape(str)
```

Comment:
Dumping to JSON gives you unescaped JSON, e.g. a hash of `{ h: { foo: 'bar' } }` would come out as `h={"foo":"bar"}` in the output. As this contains quotes it needs to be a qvalue and be written to the log as `h="{\"foo\":\"bar\"}"` (where the `\"` here is two separate characters, not a single quote character being displayed as escaped).

Also, dumping to JSON would quote string like `hello` to `"hello"` whereas it's cleaner to have them shown as `hello`, and you have to be careful with numbers as if they've been stored in TLS then the integer `2` might actually be a string, so you'd end up outputting it as `"2"` if you didn't check whether it was an ident.